cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1040 @@
+
+    if min_possible_similarity is None:
+        min_possible_similarity = (
+            1.0
+            if 0 in similarity_matrix.shape
+            else np.min(similarity_matrix[np.nonzero(similarity_matrix)])
+        )
+
+    auxiliary_input_dict: AuxiliaryTypesDict = {
+        "pred_labels": pred_labels,
+        "pred_label_probs": pred_label_probs,
+        "pred_bboxes": pred_bboxes,
+        "lab_labels": lab_labels,
+        "lab_bboxes": lab_bboxes,
+        "similarity_matrix": similarity_matrix,
+        "min_possible_similarity": min_possible_similarity,
+    }
+
+    return auxiliary_input_dict
+
+
+def _get_valid_inputs_for_compute_scores(
+    alpha: float,
+    labels: Optional[List[Dict[str, Any]]] = None,
+    predictions: Optional[List[np.ndarray]] = None,
+) -> List[AuxiliaryTypesDict]:
+    if predictions is None or labels is None:
+        raise ValueError(
+            f"Predictions and labels can not be None. Both are needed to get valid inputs."
+        )
+    min_possible_similarity = _get_min_possible_similarity(alpha, predictions, labels)
+
+    auxiliary_inputs = []
+
+    for prediction, label in zip(predictions, labels):
+        auxiliary_input_dict = _get_valid_inputs_for_compute_scores_per_image(
+            alpha=alpha,
+            label=label,
+            prediction=prediction,
+            min_possible_similarity=min_possible_similarity,
+        )
+        auxiliary_inputs.append(auxiliary_input_dict)
+
+    return auxiliary_inputs
+
+
+def _get_valid_score(scores_arr: np.ndarray, temperature: float) -> float:
+    """Given scores array, returns valid score (softmin) or 1. Checks validity of score."""
+    scores_arr = scores_arr[~np.isnan(scores_arr)]
+    if len(scores_arr) > 0:
+        valid_score = _softmin1D(scores_arr, temperature=temperature)
+    else:
+        valid_score = 1.0
+    return valid_score
+
+
+def _pool_box_scores_per_image(box_scores: List[np.ndarray], temperature: float) -> np.ndarray:
+    image_scores = np.empty(
+        shape=[
+            len(box_scores),
+        ]
+    )
+    for idx, box_score in enumerate(box_scores):
+        image_score = _get_valid_score(box_score, temperature=temperature)
+        image_scores[idx] = image_score
+    return image_scores
+
+
+def _compute_overlooked_box_scores_for_image(
+    alpha: float,
+    high_probability_threshold: float,
+    label: Optional[Dict[str, Any]] = None,
+    prediction: Optional[np.ndarray] = None,
+    pred_labels: Optional[np.ndarray] = None,
+    pred_label_probs: Optional[np.ndarray] = None,
+    pred_bboxes: Optional[np.ndarray] = None,
+    lab_labels: Optional[np.ndarray] = None,
+    lab_bboxes: Optional[np.ndarray] = None,
+    similarity_matrix: Optional[np.ndarray] = None,
+    min_possible_similarity: Optional[float] = None,
+) -> np.ndarray:
+    """This method returns one score per predicted box (above threshold) in an image. Score from 0 to 1 ranking how overlooked the box is."""
```

Comment:
future you will thank you for finalizing this documentation now (before you forget it all). Same for all the other helper methods we plan to eventually make public.  Don't let this block the PR, but i'd try to do it sooner